google-cloud-spanner/src/main/java/com/google/cloud/spanner/connection/ConnectionImpl.java

```diff
@@ -959,6 +1034,7 @@ private UnitOfWork createNewUnitOfWork() {
               .setReadOnlyStaleness(readOnlyStaleness)
               .setStatementTimeout(statementTimeout)
               .withStatementExecutor(statementExecutor)
+              .setTransactionTag(transactionTag)
```

Comment:
How will this behave if the user never set a transaction tag?

Reply:
It is nullable, so that is not a problem. It is verified by [this test case](https://github.com/googleapis/java-spanner/blob/6fcbdd022d9bf610a437f237814e4147cf1106db/google-cloud-spanner/src/test/java/com/google/cloud/spanner/connection/TaggingTest.java#L36).